### PR TITLE
Fix: #1954. No normative text in security considerations.

### LINKED ISSUE #1954: No normative statements in Security Considerations
## I expect

- no normative statements in security considerations

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- no normative text in security considerations.
Fix #1954

### CODE REVIEW

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -512,11 +512,9 @@ digest validation failure at the recipient, preventing the application from
...
 endpoints. See also {{digest-and-content-location}}.
 
-Digest fields SHOULD always be used over a connection that provides integrity at
-the transport layer that protects HTTP fields.
-
-A `Representation-Digest` field using NOT RECOMMENDED hashing algorithms SHOULD NOT be used in
-signatures.
+As explained in {{sec-limitations}}, integrity fields rely
+on transport layer integrity protection mechanisms,
+while {{algorithms}} provides further guidance on algorithm's choice.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:517]: Thanks.

Seeing it like this, is there even a need to say anything here? The referenced sections apply to the usage of digest and don't appear to add any further insight or considerations when used with Signature. 

@jricher WDYT about just deleting the two sentences and not adding anything more?

Comment (jricher) [reply draft-ietf-httpbis-digest-headers.md:517]: I actually like the cross-reference here, I would keep it this way.

**Review by jricher — APPROVED** [review]

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:517]: Interesting. If someone is using Digests with signature, they should really apply all the recommondations and apply all the considerations of the document. There seems to me no need to call out any particular section as unique to Signatures, unless there is something to say about why. Is that the case here?

The rest of this section _does_ achieve the goal of signature-specific considerations. It's just these that stick out to me.

Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:517]: @LPardue  {{algorithms}} actually references a signature-related normative text, so I think it should stay. The TLS part in 2022 is probably redundant for a careful reader.

Can we agree on removing the transport layer text and leave the {{algorithms}} reference? cc: @jricher

Comment (jricher) [reply draft-ietf-httpbis-digest-headers.md:517]: I still think they can both stay, but at this stage I'll leave it up to your discretion if you think it is clearer without them.

Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:517]: So reading the text in {{algorithms}} the relevant part seems to be about saying signatures is a potential adversarial setting. 

But we already have the Algorithm Agility security consideration section, which now contains nice text about considerations for choosing secure or insecure algorithms.

So instead I would say

"Signatures are likely to be deemed an adversarial setting when applying integrity digests; see {{algorithm agility}}."

And still drop the part about transport integrity.

Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:517]: deal!

**Review by LPardue — APPROVED** [review]

**Pull request merged**